base/settings/R/read.settings.R

```diff
@@ -97,3 +100,39 @@ read.settings <- function(inputfile = "pecan.xml") {
 
   return(invisible(settings))
 }
+
+
+
+#' Strip comments from parsed pecan.xml
+#'
+#' Allows use of HTML style comments (`<!-- a comment -->`) in pecan.xml files
+#' by removing them after converted to a list. Inspired by
+#' https://stackoverflow.com/questions/37853679/removing-elements-in-a-nested-r-list-by-name
+#'
+#' @param x a settings list
+#'
+#' @return list
+#' @noRd
+#'
+strip_comments <- function(x) {
+  # function to find depth of a list element
+  # see http://stackoverflow.com/questions/13432863/determine-level-of-nesting-in-r
+  depth <- function(this, thisdepth = 0) {
+    if (!is.list(this)) {
+      return(thisdepth)
+    } else{
+      return(max(unlist(
+        lapply(this, depth, thisdepth = thisdepth + 1)
+      )))
+    }
+  }
+  
+  thisdepth <- depth(x)
+  if (thisdepth == 0) {
+    return(x)
+  } else if (length(nameIndex <- which(names(x) == "comment"))) {
```

Comment:
Matter of style, but I’d hoist the assignment to nameIndex up to between current  lines 130 and 131